src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,35 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final int DATE = 0;
+    private static final int NAME = 1;
+    private static final int HOURS = 2;
+    private static final int RATE = 3;
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder report = new StringBuilder("Report for period " + dateFrom + " - " + dateTo);
+        final LocalDate beginning = LocalDate.parse(dateFrom, FORMATTER);
+        final LocalDate end = LocalDate.parse(dateTo, FORMATTER);
+        int income = 0;
```

Comment:
The variable 'income' should have a more informative name that reflects its purpose, such as 'totalIncome' or 'employeeIncome', to make the code more readable and understandable.